packages/netlify-cms-widget-relation/src/RelationControl.js

```diff
@@ -128,24 +128,38 @@ export default class RelationControl extends React.Component {
   parseHitOptions = hits => {
     const { field } = this.props;
     const valueField = field.get('valueField');
+    const searchFields = field.get('searchFields');
     const displayField = field.get('displayFields') || field.get('valueField');
 
-    return hits.map(hit => {
-      let labelReturn;
-      if (List.isList(displayField)) {
-        labelReturn = displayField
-          .toJS()
-          .map(key => this.parseNestedFields(hit, key))
-          .join(' ');
-      } else {
-        labelReturn = this.parseNestedFields(hit, displayField);
-      }
-      return {
-        data: hit.data,
-        value: this.parseNestedFields(hit, valueField),
-        label: labelReturn,
-      };
-    });
+    let data = hits;
+    if (field.get('file')) {
+      data = get(hits, `[0].data.${searchFields}`, []).map(item => ({ data: item }));
```

Comment:
Not sure I understand why the field that will be searched for in the file collection is assumed to be an array.

Reply:
You're right, I should add an array check here, thanks for spotting this!